rfcs/address_space_overrides.md

```diff
@@ -0,0 +1,111 @@
+# RFC #72: Address space overrides
+
+## Summary
+
+The goal of this RFC is to enable web platform tests to cover the
+[CORS-RFC1918](https://wicg.github.io/cors-rfc1918) specification entirely.
+More specifically, to enable tests to exercise user agent behavior in the face
+of responses served from `local`, `private` and `public`
+[address spaces](https://wicg.github.io/cors-rfc1918#address-space).
+
+This is achieved by passing command-line flags to the browser under test forcing
+it to artificially consider specific IP addresses `local`, `private` or
+`public`.
```

Comment:
What about:
have a command line flag or pref that overwrites an address. Browsers would overwrite the IP address just before creating a socket. For example map 100.10.10.10 to 127.0.0.1. The rest of the browser will only see 100.10.10.10, and only the socket will be created using 127.0.0.1

I think the approach will be less promte to implementation errors because it is simpler to implement.

Reply:
Interesting! Thanks for reviewing.

That sounds OK to me! As long as the socket then reports back, when asked, that it is connected to the fake address, then I think it works. It would be injecting logic deep into the network stack for a feature that is otherwise situated at a layer far above the network.

It's not clear to me that this would be easier to implement, since it requires persisting the faked state somewhere in the network stack. The approach I suggested instead only requires injecting a lookup in some override map at address space computation time. That computation should already happen in some CORS-RFC1918-specific code.

WDYT?

Reply:
@ddragana friendly ping :)

Reply:
@Hexcles this would also alter the meaning of specific IP addresses in the context of the test, but not for the whole system. Do you have similar reservations about this approach as to the `/etc/hosts` approach or would this be fine with you?

Reply:
This approach doesn't seem to require root AFAICT, so it seems fine, but let me pass it to @stephenmcgruer as I'm no longer working on WPT.

Reply:
Apologies for the delay; yes, from the WPT perspective I think that would be fine as it fits our preferences (no root, pass a command line flag). I think the big question would still be one of persuading other browser vendors :)

Reply:
Thanks! I just need a signal from @ddragana now to fix up this PR and get to work :)

Reply:
Added a section to describe this alternative. @ddragana can you please take a look?

Reply:
Friendly ping @ddragana - I'd like to get this RFC finalized.

Reply:
Sorry, I have missed the ping.
It looks fine.

Reply:
Thank you, resolving.